ddcci-multi-monitor@tim-we/files/ddcci-multi-monitor@tim-we/applet.js

```diff
@@ -188,6 +195,46 @@ class DDCMultiMonitor extends Applet.IconApplet {
             this.updateMenu();
         }
     }
+
+    // Change the brightness when scrolling on the icon
+    _onScrollEvent(actor, event) {
+        let direction = event.get_scroll_direction();
+        if (direction == Clutter.ScrollDirection.SMOOTH) { // Prevents being triggered twice
+            return
+        }
+
+        if (direction == Clutter.ScrollDirection.DOWN) {
+            this.monitors.forEach((monitor) => {
+                monitor.brightness = Math.max(0, monitor.brightness - BRIGHTNESS_ADJUSTMENT_STEP); // Decrease brightness by the step amount, making sure to not go below 0
+                // Tried making this into a function, it did not go well
+                this.set_applet_tooltip("Brightness: " + monitor.brightness + "%"); 
```

Comment:
> Tried making this into a function, it did not go well

What do you mean by this?

I don't think its a good idea to set this in the monitor loop as the tooltip will then show only the brightness of the last monitor. I suggest you use something like 
```js
const tooltipMessage = this.monitors.map(monitor => {
    monitor.brightness = /* ... */
   /* ... */
   return `${monitor.name}: ${monitor.brightness}%`;
}).join("\n");

to display all brightness values for all monitors.

Reply:
> What do you mean by this?

I should have been more specific, I meant all of lines 210 to 218. For some reason when I tried to put it into a function, the function would just not get called. It doesn't help that I am not great at javascript.

> I don't think its a good idea to set this in the monitor loop

I completely agree, thanks for the suggestion.
I'm not sure where or how exactly to add the code you suggested, but here works (at least for 1 display):

```js
if (direction == Clutter.ScrollDirection.DOWN) {
    let tooltipMessage = this.monitors.map(monitor => {
        monitor.brightness = Math.max(0, monitor.brightness - BRIGHTNESS_ADJUSTMENT_STEP);
        setTimeout(() => { // Prevent weirdness when scrolling multiple steps
            monitor.setBrightness(monitor.brightness);
        }, 200);
        return `${monitor.name}: ${monitor.brightness}%`;
    }).join("\n");

    this.set_applet_tooltip(tooltipMessage)
    this._applet_tooltip.show()
    setTimeout(() => {
        this._applet_tooltip.hide();
        this.set_applet_tooltip(DEFAULT_TOOLTIP);
    }, 2500);
    
}

else if (direction == Clutter.ScrollDirection.UP) {
    let tooltipMessage /* ... */
```

Should I push this into the pull request, or did you have something else in mind?